lib/gems/pending/util/mount/miq_local_mount_session.rb

```diff
@@ -0,0 +1,48 @@
+require 'util/mount/miq_generic_mount_session'
+
+# MiqLocalMountSession is meant to be a representation of the local file system
+# that conforms to the same interface as MiqLocalMountSession (and by proxy,
+# MiqFileSystem::Interface).
+#
+# See MiqGenericMountSession for info on methods available.
+class MiqLocalMountSession < MiqGenericMountSession
+  def self.uri_scheme
+    "file".freeze
+  end
+
+  # no-op these since they are not relavent to the local file system
+  #
+  # rubocop:disable Style/SingleLineMethods, Layout/EmptyLineBetweenDefs
+  def connect; end      # :nodoc:
+  def disconnect; end   # :nodoc:
+  def mount_share; end  # :nodoc:
+  # rubocop:enable Style/SingleLineMethods, Layout/EmptyLineBetweenDefs
+
+  def relative_to_mount(dest_uri) # :nodoc:
+    dest_uri
+  end
+
+  def uri_to_local_path(dest_uri) # :nodoc:
+    File.expand_path(dest_uri)
+  end
+
+  # :nodoc:
+  #
+  # This is an override to allow methods that make use of `@mnt_point` to
+  # behave as we would expect them to.
+  #
+  # Determine the mount point relative to the destination file, and just remove
+  # the portions of the expanded path that already exist in the
+  # `remote_file_path` (might be everything if `remote_file_path` is a absolute
+  # path)
+  def mnt_point
```

Comment:
This logic looks a bit complicated. Should we test this directly or you you think this is exercised enough through the existing tests?

Reply:
It is exercised (I believe) in the `MiqLocalMountSession` tests, as I switch up what I pass to `.add` in each of the different tests.

But I can target it specifically in that spec if you want (seems like it wouldn't be a bad idea).

Reply:
Actually, after doing some investigating, I am going to try deleting this code and use a slightly different approach.  I will report back, but that might be the easier route.